Snek.cs

```diff
@@ -1,131 +1,139 @@
 using System.Numerics;
 using System.Reflection;
 using System.Runtime.CompilerServices;
 using Raylib_cs;
 
 namespace Snek;
 
+public struct Cord
+{
+    public int X;
+    public int Y;
+
+    // TODO: not really needed, see Grid.
+    public Cord(int x, int y)
+    {
+        X = x;
+        Y = y;
+    }
+
+    public readonly bool Equals(Cord other)
+    {
+        return (this.X == other.X) && (this.Y == other.Y);
+    }
+}
+
 public class Snek
 {
     public enum Direction
     {
-        Up = 3,
-        Down = 1,
-        Left = 2,
-        Right = 0,
+        Right,
+        Down,
+        Left,
+        Up
     }
 
     // starting position and direction of the snake
-    public int snakeRow = 10;
-    public int snakeCol = 10;
+    public Cord[] snakeCords;
     public Direction direction = Direction.Right;
 
     // direction in which it can move
     int[,] D = { { 0, 1 }, { 1, 0 }, { 0, -1 }, { -1, 0 } };
 
-    // storing the last position in the arrays based on top score
-    public int[] prevLocRow;
-    public int[] prevLocCol;
-
     public Snek(int topScore)
     {
-        prevLocRow = new int[topScore];
-        prevLocCol = new int[topScore];
+        snakeCords = new Cord[topScore + 1];
+        snakeCords[0] = new Cord(10, 10);
+    }
+
+    // snek properties for row and col 
+    public Cord Head
+    {
+        get { return snakeCords[0]; }
+        set { snakeCords[0] = value; }
     }
 
     // methods now
-    public void Move(int gridRows, int gridCols)
+    public void Move(Grid grid)
     {
         //keeps track of all of the snake
-        int a = prevLocRow.Length - 2;
+        int a = snakeCords.Length - 2;
         while (a >= 0)
 
         {
-            prevLocRow[a + 1] = prevLocRow[a];
-            prevLocCol[a + 1] = prevLocCol[a];
+            snakeCords[a + 1] = snakeCords[a];
             a = a - 1;
         }
 
-        prevLocRow[0] = snakeRow;
-        prevLocCol[0] = snakeCol;
-        snakeRow += D[(int)direction, 0];
-        snakeCol += D[(int)direction, 1];
+        //endless faff of using structs fml 
+        Cord newHead = Head;
+
+        newHead.X += D[(int)direction, 0];
+        newHead.Y += D[(int)direction, 1];
 
         // wrapping around situ
-        if (snakeRow >= gridRows){snakeRow = 0;}
-        if (snakeRow < 0){snakeRow = gridRows - 1;}
-        if (snakeCol >= gridCols){snakeCol = 0;}
-        if (snakeCol < 0){snakeCol = gridCols - 1;}
+        if (newHead.X >= grid.cols) { newHead.X = 0; }
+        if (newHead.X < 0) { newHead.X = grid.cols - 1; }
+        if (newHead.Y >= grid.rows) { newHead.Y = 0; }
+        if (newHead.Y < 0) { newHead.Y = grid.rows - 1; }
 
+        Head = newHead;
     }
 
     public void SetDirection(Direction direction)
     {
         this.direction = direction;
     }
 
-    public bool Overlaps(int row, int col, int score, bool skipHead = false)
+    public bool Overlaps(Cord newSnakeCords, int score, bool skipHead = false)
```

Comment:
```suggestion
    public bool Overlaps(Cord coords, int score, bool skipHead = false)
```

This method should be independent of any use case, naming the variable newsnakecoords is implying you'll only use it to ever compare the new snake coords. It should be able to compare any coordinate passed.